src/main/java/org/folio/inventory/consortium/services/ConsortiumServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+  private static final String SHARE_INSTANCE_ENDPOINT = "/consortia/%s/sharing/instances";
+  private static final String SHARING_INSTANCE_ERROR = "Error during sharing Instance for sourceTenantId: %s, targetTenantId: %s, instanceIdentifier: %s, status code: %s, response message: %s";
+  private final HttpClient httpClient;
+
+  public ConsortiumServiceImpl(HttpClient httpClient) {
+    this.httpClient = httpClient;
+  }
+
+  @Override
+  public Future<SharingInstance> createShadowInstance(Context context, String instanceId, ConsortiumConfiguration consortiumConfiguration) {
+    Context centralTenantContext = constructContext(consortiumConfiguration.getCentralTenantId(), context.getToken(), context.getOkapiLocation());
+    SharingInstance sharingInstance = new SharingInstance();
+    sharingInstance.setSourceTenantId(consortiumConfiguration.getCentralTenantId());
+    sharingInstance.setInstanceIdentifier(UUID.fromString(instanceId));
+    sharingInstance.setTargetTenantId(context.getTenantId());
+    return shareInstance(centralTenantContext, consortiumConfiguration.getConsortiumId(), sharingInstance);
+  }
+
+  @Override
+  public Future<Optional<ConsortiumConfiguration>> getConsortiumConfiguration(Context context) {
+    CompletableFuture<Optional<ConsortiumConfiguration>> completableFuture = createOkapiHttpClient(context)
+      .thenCompose(client ->
+        client.get(context.getOkapiLocation() + USER_TENANTS_ENDPOINT).toCompletableFuture()
+          .thenCompose(httpResponse -> {
+            if (httpResponse.getStatusCode() == HttpStatus.SC_OK) {
+              JsonArray userTenants = httpResponse.getJson().getJsonArray("userTenants");
+              if (userTenants.isEmpty()) {
+                LOGGER.debug("getCentralTenantId:: Central tenant and consortium id not found");
+                return CompletableFuture.completedFuture(Optional.empty());
+              }
+              String centralTenantId = userTenants.getJsonObject(0).getString("centralTenantId");
+              String consortiumId = userTenants.getJsonObject(0).getString("consortiumId");
+              LOGGER.debug("getCentralTenantId:: Found centralTenantId: {} and consortiumId: {}", centralTenantId, consortiumId);
+              return CompletableFuture.completedFuture(Optional.of(new ConsortiumConfiguration(centralTenantId, consortiumId)));
+            } else {
+              String message = String.format("Error retrieving centralTenantId by tenant id: %s, status code: %s, response message: %s",
+                context.getTenantId(), httpResponse.getStatusCode(), httpResponse.getBody());
+              LOGGER.warn(String.format("getCentralTenantId:: %s", message));
+              return CompletableFuture.failedFuture(new ConsortiumException(message));
+            }
+          }));
+    return Future.fromCompletionStage(completableFuture);
+  }
+
+  // Returns completed future if the sharing status is "IN_PROGRESS" or "COMPLETE"
```

Comment:
returns "successful" future, not just completed.

Reply:
Done, Thanks